Refactor: use cdt_identity `logout` view

lalver1 opened the pull request:
Closes #2787

As part of the oauth refactor, this PR replaces `benefits.oauth.views.logout` with `cdt_identity.views.logout`.

## Reviewing

Run through a login.gov flow and ensure you can confirm your eligibility. On the eligibility confirmation page, click on `Sign out of Login.gov ` and check that the `cdt_identity` `logout` view is called.  In addition, a `started sign out` analytics event will be created as part of the `pre_logout` hook; and a `finished sign out` analytics event will be created as part of the `post_logout` hook.

angela-tran commented on tests/pytest/oauth/test_views.py:
```diff
@@ -161,38 +158,12 @@ def test_cancel_no_session_flow(client):
+    response = client.get(path)
 
-    assert not session.logged_in(app_request)
-    assert session.enrollment_token(app_request) is False
-    assert session.logged_in(app_request) is False
-    assert OAuthSession(app_request).claims_result == ClaimsResult()
```
_(sorry, I know this PR is still WIP but just thought I'd go ahead and post this suggestion :sweat_smile:)_

Can we add these assertions to [`test_hooks.test_pre_logout`](https://github.com/cal-itp/benefits/blob/main/tests/pytest/oauth/test_hooks.py#L52)?

lalver1 replied:
Not at all! This is great, I just pushed a small refactor of the `test_xxx_no_session_flow` tests but I'll look at this next 👍

angela-tran commented on tests/pytest/oauth/test_views.py:
```diff
@@ -109,8 +106,12 @@ def test_login_oauth_client_init_error(client, mocked_hook_analytics_module, moc
+@pytest.mark.parametrize(
+    "route",
+    [routes.OAUTH_LOGIN, routes.OAUTH_AUTHORIZE, routes.OAUTH_CANCEL, routes.OAUTH_LOGOUT, routes.OAUTH_POST_LOGOUT],
+)
```
:raised_hands:

angela-tran commented:
Just realized this is still in draft. @lalver1 are there changes you're still wanting to make on this PR?

lalver1 commented:
It's ready to go @angela-tran! I only wanted to clarify if the few lines I added in https://github.com/cal-itp/benefits/pull/2819/files#diff-c4e66e5bbdbde74f5062e2a19cb1b0374dc4180b1de7f0b932e169f60ef25e02R53 were what you had in mind (but that's part of the review 😅 ). I think the reason to add them is because we removed `test_logout` yet we still want to have these assertions.

angela-tran commented:
> It's ready to go @angela-tran! I only wanted to clarify if the few lines I added in https://github.com/cal-itp/benefits/pull/2819/files#diff-c4e66e5bbdbde74f5062e2a19cb1b0374dc4180b1de7f0b932e169f60ef25e02R53 were what you had in mind (but that's part of the review 😅 ). 

Yep, it looks great!

> I think the reason to add them is because we removed `test_logout` yet we still want to have these assertions.

Yep, that was my thinking. Ideally those assertions would've been a part of the [test](https://github.com/cal-itp/benefits/pull/2769/files#diff-c4e66e5bbdbde74f5062e2a19cb1b0374dc4180b1de7f0b932e169f60ef25e02R39) introduced in #2769, but you know, hindsight is 20/20.
